README.md

```diff
@@ -1,64 +1,148 @@
 ```shell
-# Regular Tests
 npx hardhat test
+```
 
-# Bundler Integration Tests
-# Install realpath
+For Bundler Integration Tests, first install `realpath`:
+```shell
 brew install coreutils
-# Run Bundler Integration Tests
+```
+
+Then, run the Bundler Integration Tests:
+```shell
 yarn bundler-test
+```
 
-# other
-npx hardhat accounts
+### 📦 Compilation & Deployment
+
+Compile contracts:
+```shell
 npx hardhat compile
+```
+
+Clean the environment:
+```shell
 npx hardhat clean
+```
+
+Start a local Ethereum node:
+```shell
 npx hardhat node
+```
+
+Deploy contracts:
+```shell
+npx hardhat run scripts/deploy.ts
+TS_NODE_FILES=true npx ts-node scripts/deploy.ts
+```
+
+### 📈 Analysis & Reporting
+
+Display available accounts:
+```shell
+npx hardhat accounts
+```
+
+Get help on Hardhat commands:
+```shell
 npx hardhat help
+```
+
+Test with gas report:
+```shell
 REPORT_GAS=true npx hardhat test
+```
+
+Generate code coverage report:
+```shell
 npx hardhat coverage
-npx hardhat run scripts/deploy.ts
-TS_NODE_FILES=true npx ts-node scripts/deploy.ts
+```
+
+### 🧹 Code Quality & Formatting
+
+Lint JavaScript and TypeScript files:
+```shell
 npx eslint '**/*.{js,ts}'
+```
+
+Automatically fix linting issues:
+```shell
 npx eslint '**/*.{js,ts}' --fix
+```
+
+Check formatting for JSON, Solidity, and Markdown files:
+```shell
 npx prettier '**/*.{json,sol,md}' --check
+```
+
+Automatically format files:
+```shell
 npx prettier '**/*.{json,sol,md}' --write
+```
+
+Lint Solidity contracts:
+```shell
 npx solhint 'contracts/**/*.sol'
+```
+
+Automatically fix issues in Solidity contracts:
+```shell
 npx solhint 'contracts/**/*.sol' --fix
 ```
+---
+
+This format separates the description from the command, making it clearer and more readable.
+
+## 🔍 Etherscan Verification
+
+To verify on Etherscan, deploy a contract to an Ethereum network supported by Etherscan, like Ropsten. Set up your `.env` file, deploy your contract, and then verify:
+
+```shell
+hardhat run --network ropsten scripts/deploy.ts
+npx hardhat verify --network ropsten DEPLOYED_CONTRACT_ADDRESS "Hello, Hardhat!"
+```
+
+## ⚡ Performance Optimizations
+
+Boost your tests and scripts' speed by setting the `TS_NODE_TRANSPILE_ONLY` environment variable to `1` in Hardhat's environment. More details are available in the [documentation](https://hardhat.org/guides/typescript.html#performance-optimizations).
 
-# Performance optimizations
+---
+## 🤝 Contributing
 
-For faster runs of your tests and scripts, consider skipping ts-node's type checking by setting the environment variable `TS_NODE_TRANSPILE_ONLY` to `1` in hardhat's environment. For more details see [the documentation](https://hardhat.org/guides/typescript.html#performance-optimizations).
+Biconomy Smart Account is an open-source project. Contributions are welcome. If you're interested in contributing, please check our [contribution guidelines](./CONTRIBUTING.md) and feel free to submit pull requests or raise issues.
```

Comment:
There is no CONTRIBUTING.md file in this repo. This link leads to 404 page.

Reply:
yes I know, that's related to what we discussed earlier about having a proper guideline for contribution. But I put it here for the future